app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,554 @@
+                                newPage.getUnpublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+
+                            if (newPage.getPublishedPage() != null
+                                    && newPage.getPublishedPage().getLayouts() != null) {
+
+                                newPage.getPublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+                        });
+                        applicationJson.setPageList(newPageList);
+                        applicationJson.setMongoEscapedWidgets(mongoEscapedWidgetsNames);
+                        return datasourceRepository
+                            .findAllByOrganizationId(organizationId, AclPermission.MANAGE_DATASOURCES)
+                            .collectList();
+                    })
+                    .flatMapMany(datasourceList -> {
+                        datasourceList.forEach(datasource ->
+                            datasourceIdToNameMap.put(datasource.getId(), datasource.getName()));
+                        
+                        applicationJson.setDatasourceList(datasourceList);
+                        return newActionRepository
+                            .findByApplicationId(applicationId, AclPermission.MANAGE_ACTIONS, null);
+                    })
+                    .collectList()
+                    .map(newActionList -> {
+                        Set<String> concernedDBNames = new HashSet<>();
+                        newActionList.forEach(newAction -> {
+                            newAction.setPluginId(pluginMap.get(newAction.getPluginId()));
+                            newAction.setOrganizationId(null);
+                            newAction.setPolicies(null);
+                            newAction.setApplicationId(null);
+                            //Collect Datasource names to filter only required datasources
+                            if (newAction.getPluginType() == PluginType.DB || newAction.getPluginType() == PluginType.API) {
+                                concernedDBNames.add(mapDatasourceIdToNewAction(newAction.getPublishedAction(), datasourceIdToNameMap));
+                                concernedDBNames.add(mapDatasourceIdToNewAction(newAction.getUnpublishedAction(), datasourceIdToNameMap));
+                            }
+                            if (newAction.getUnpublishedAction() != null) {
+                                ActionDTO actionDTO = newAction.getUnpublishedAction();
+                                actionDTO.setPageId(pageIdToNameMap.get(actionDTO.getPageId()));
+                            }
```

Comment:
Do we need to do this for published action as well?

```suggestion
                            if (newAction.getUnpublishedAction() != null) {
                                ActionDTO actionDTO = newAction.getUnpublishedAction();
                                actionDTO.setPageId(pageIdToNameMap.get(actionDTO.getPageId()));
                            }
                            if (newAction.getPublishedAction() != null) {
                                ActionDTO actionDTO = newAction.getPublishedAction();
                                actionDTO.setPageId(pageIdToNameMap.get(actionDTO.getPageId()));
                            }
```

Reply:
Do we have different pageIds for published and unpublished actionDTOs? If not this has been handled in import code.

Reply:
Ah I see. Yeah both will have the same pageId, but I was thinking this will leak an ID value into the export file right? That also doesn't sound like a big problem. Yeah fine, if it's handled at import time, we should be okay then.